Rakefile

```diff
@@ -9,6 +9,7 @@ Rake::TestTask.new(:test) do |t|
   t.libs << "test"
   t.libs << "lib"
   t.test_files = FileList["test/**/*_test.rb"]
+  t.warning = false
```

Comment:
Which warnings is this suppressing? Should we fix these instead?

Reply:
These three:

```
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.8/lib/active_support/core_ext/time/deprecated_conversions.rb:42: warning: method redefined; discarding old to_s
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.8/lib/active_support/time_with_zone.rb:210: warning: previous definition of to_s was here
```


```
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.11/lib/zeitwerk/kernel.rb:38: warning: /Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.11/lib/zeitwerk/kernel.rb:38: warning: loading in progress, circular require considered harmful - /Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/yard-0.9.34/lib/yard.rb
```

```
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/date_time_parser.rb:837: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/date_time_parser.rb:691: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32457: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32618: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32654: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32743: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32759: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32825: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32866: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:32891: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/address_lists_parser.rb:31987: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_disposition_parser.rb:795: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_disposition_parser.rb:819: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_disposition_parser.rb:586: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:761: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:592: warning: assigned but unused variable - disp_type_s
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:607: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_transfer_encoding_parser.rb:358: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_type_parser.rb:942: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_type_parser.rb:966: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_type_parser.rb:712: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/envelope_from_parser.rb:3241: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/message_ids_parser.rb:5112: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/message_ids_parser.rb:4848: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/mime_version_parser.rb:322: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/phrase_lists_parser.rb:702: warning: assigned but unused variable - testEof
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:7861: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8058: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8091: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8108: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8215: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8313: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8329: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8345: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8419: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8437: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8455: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8541: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8568: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8614: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8647: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8665: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8685: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8707: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:8727: warning: statement not reached
/Users/joelhawksley/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/received_parser.rb:7514: warning: assigned but unused variable - testEof
```

I don't see an obvious resolution for any of them.

Reply:
That's unfortunate since I'd like to still see warnings for ViewComponent related code. I did find https://github.com/jeremyevans/ruby-warning but not sure if it's worth the effort. Thanks for the context!

Reply:
Oh how cool! `ruby-warning` was totally worth the minimal effort. Just pushed the change ❤️